doc/run-partial.md

```diff
@@ -1,3 +1,82 @@
-# axe.runPartial & axe.finishRun
+# axe runPartial / finishRun
 
-TODO
+`axe.runPartial` and `axe.finishRun` are two methods which allow axe to test a page in two stages. This is different from `axe.run()`, in that `axe.runPartial` and `axe.finishRun` does not require communication between frames. This can be useful when frame communication is not possible or not secure. See [axe.frameMessenger](frame-messenger.md) for information on frame communication.
+
+To use these methods, call `axe.runPartial()` in the top window, and in all nested frames and iframes. The results are then put into an array, and then passed to `axe.finishRun()`. `axe.finishRun()` then complete the test and generates the axe report.
+
+This results in code that looks something like this:
+
+```js
+const partialResults = await Promise.all(
+  runPartialRecursive(context, options)
+)
+const axeResults = await axe.finishRun(partialResults, options);
+```
+
+**note**: The code in this page uses native DOM methods. This will only work on frames with the same origin. Scripts do not have access to `contentWindow` of cross-origin frames. Use of `runPartial` and `finishRun` in browser driver like Selenium and Puppeteer works in the same way.
```

Comment:
```suggestion
**note**: The code in this page uses native DOM methods. This will only work on frames with the same origin. Scripts do not have access to `contentWindow` of cross-origin frames. Use of `runPartial` and `finishRun` in browser drivers like Selenium and Puppeteer works in the same way.
```